packages/pass-style/src/types.d.ts

```diff
@@ -0,0 +1,182 @@
+  (p: any[]): 'copyArray';
+  (p: Iterable<any>): 'remotable';
+  (p: Iterator<any, any, undefined>): 'remotable';
+  <T extends PassStyled<any>>(p: T): ExtractStyle<T>;
+  (p: { [key: string]: any }): 'copyRecord';
+  (p: any): PassStyle;
+};
+/**
+ * A Passable is PureData when its entire data structure is free of PassableCaps
+ * (remotables and promises) and error objects.
+ * PureData is an arbitrary composition of primitive values into CopyArray
+ * and/or
+ * CopyRecord and/or CopyTagged containers (or a single primitive value with no
+ * container), and is fully pass-by-copy.
+ *
+ * This restriction assures absence of side effects and interleaving risks *given*
+ * that none of the containers can be a Proxy instance.
+ * TODO SECURITY BUG we plan to enforce this, giving PureData the same security
+ * properties as the proposed
+ * [Records and Tuples](https://github.com/tc39/proposal-record-tuple).
+ *
+ * Given this (currently counter-factual) assumption, a PureData value cannot
+ * be used as a communications channel,
+ * and can therefore be safely shared with subgraphs that should not be able
+ * to communicate with each other.
+ * Without that assumption, such a guarantee requires a marshal-unmarshal round
+ * trip (as exists between vats) to produce data structures disconnected from
+ * any potential proxies.
+ */
+export type PureData = Passable<never, never>;
+export type TaggedRecord<
+  S extends string,
+  I extends InterfaceSpec,
+> = PassStyled<S> & {
+  [Symbol.toStringTag]: I;
+};
+/**
+ * An object marked as remotely accessible using the `Far` or `Remotable`
+ * functions, or a local presence representing such a remote object.
+ *
+ * A more natural name would be Remotable, but that could be confused with the
+ * value of the `Remotable` export of this module (a function).
+ */
+export type RemotableObject<I extends string = string> = TaggedRecord<
+  'remotable',
+  I
+>;
+/**
+ * The authority-bearing leaves of a Passable's pass-by-copy superstructure.
+ */
+export type PassableCap = Promise<any> | RemotableObject;
+/**
+ * A Passable sequence of Passable values.
+ */
+export type CopyArray<T extends Passable = any> = Array<T>;
+
+/**
+ * A Passable dictionary in which each key is a string and each value is Passable.
+ */
+export type CopyRecord<T extends Passable = any> = Record<string, T>;
+/**
+ * A Passable "tagged record" with semantics specific to the tag identified in
+ * the `[Symbol.toStringTag]` property (such as 'copySet', 'copyBag',
+ * or 'copyMap').
+ * It must have a property with key equal to the `PASS_STYLE` export and
+ * value 'tagged'
+ * and no other properties except `[Symbol.toStringTag]` and `payload`.
+ */
+export type CopyTagged<
+  Tag extends string = string,
+  Payload extends Passable = any,
+> = TaggedRecord<'tagged', Tag> & {
+  payload: Payload;
+};
+/**
+ * This is an interface specification.
+ * For now, it is just a string, but will eventually be `PureData`. Either
```

Comment:
No change suggested.

But noting that I doubt we ever would change this from a string. But the time we'd want to, it will be too great a compat hazard to bother with. Unless we encounter a compelling need, which I do not expect.

Reply:
I'll revise the text to something less certain than "will eventually be"